crypto/bn/bn_sqrt.c

```diff
@@ -179,7 +179,7 @@ BIGNUM *BN_mod_sqrt(BIGNUM *in, const BIGNUM *a, const BIGNUM *p, BN_CTX *ctx)
             if (!BN_set_word(y, i))
                 goto end;
         } else {
-            if (!BN_rand(y, BN_num_bits(p), 0, 0))
+            if (!BN_priv_rand(y, BN_num_bits(p), 0, 0))
```

Comment:
I'm not sure whether I always understand the rationale behind your choice of public vs. private rng.

Reply:
Anything that gets send to a peer is public, the rest is private.